API.md

```diff
@@ -90,3 +90,65 @@ _Note: The `message` field is meant only to describe the error to a user, this t
 
 `{"error":{"id":"no_login","message":"You are not logged in to an account"},"status":"error"}`
 
+### `/resetpass` - Reset a password for a user
+#### Params
+- `account`
+- `email`
+
+#### Responses
+##### Success
+
+`{"status":"ok"}`
+
+#### Errors
+
+`{"error":{"id":"no_account","message":"Unable to find matching account"},"status":"error"}`
+
+`{"error":{"id":"mail_failed","message":"Unable to send reset email"},"status":"error"}`
+
+### `/resetpass/confirm` - Confirm a password reset
+This has two main uses:
+- Simply confirming the reset request and logging in the user, which will then mark the account as unconfirmed until the user changes their password, or
+- Confirming the reset request and also setting the password to `newpass` in one request (requires sending the `newpass` parameter
+
+This allows finer control from the API consumer side as to the separation of authentication and modification of the account
+
+#### Params
+- `account` - Account name to reset the password for
+- `code` - Confirmation code from the user
+- `newpass` _optional_ - If set, sets the new password in this request, otherwise a later request to `/user/set/password` is required.
+
+#### Responses
+##### Success
+
+`{"status":"ok","password_set":true|false,"account":"<ACCOUNT_NAME>","verified":true|false,"session":"<SESSION_ID>"}`
+
+- `password_set` - Was the password changed in this request or is a future request to `/user/set/password` required
+- `verified` - Is the account a verified account or not
```

Comment:
why does the status of verification need to be sent in this?

Reply:
Because as stated earlier on, if the newpass isn't sent, the account is marked unverified pending the user setting the password with user/set/password, to mimic the anope functionality

Reply:
That does not change the fact that this does not need that information passed back.

Reply:
This is meant to behave the same as the login endpoint, but with the additional information for a password reset, since this endpoint logs the user in with the temporary code anope sends out.

Reply:
Why does it need to behave the same? Its not a login endpoint. Its a password reset endpoint. Exposing information not required to complete the task at hand is a security issue. Components do not need more information than what is required to make a process happen. Also, having backend things that change based on some arbitrary behavior in an 'unrelated' API endpoint is a marker of a bad API. If you want the ability to mark a user as unverified, add an endpoint for it.

Reply:
I may just remove the ability to not set the password at that point and just keep the set/password endpoint for changing the password in a normal logged in account

Reply:
Resolved by b5186745fa7d0415ff92250d613457e3024bbcc7

Reply:
That sounds pretty clean to me